ci/build.sh

```diff
@@ -19,7 +19,7 @@ set -ex
 # for compilation or for tests in our package.yaml file (these sorts of issues
 # don't test to get noticed until after releasing otherwise).
 
-STACK="stack --no-terminal --jobs=2"
+STACK="stack --no-terminal --haddock --jobs=2"
```

Comment:
We have gone back and forth on this a couple of times, so I think we should add a comment explaining why it is necessary so that it doesn't get removed again later to speed CI up. For example:

> We test with `--haddock` because haddock generation can fail if there is invalid doc-comment syntax, and these failures are very easy to miss otherwise.

Reply:
I added comment.